AK/StringUtils.cpp

```diff
@@ -401,6 +401,18 @@ Vector<size_t> find_all(StringView haystack, StringView needle)
     return positions;
 }
 
+Vector<size_t> find_all(StringView haystack, Function<bool(char)> const& predicate)
```

Comment:
Did you consider making `Function<...>` instead be a deduced argument to avoid the `Function` construction when passed a lambda? If you are concerned about compiler messages you could also add concepts.

Deduced type:
```cpp
Vector<size_t> find_all(StringView haystack, auto && predicate)
```

Using concepts to constrain it and get better error messages would be something like:
```cpp
template<class TCallable>
requires (IsFunctionObject<TCallable> && IsCallableWithArguments<char>)
Vector<size_t> find_all(StringView haystack, TCallable && predicate)
```

Reply:
This would work for sure, but I did it this way for consistency with the way `StringView::split_view_if` does it. (which imports AK/Function.h too)
What should be done?

Reply:
I don't think it _needs_ to happen. Just wondering if it was considered. Sounds like you considered and went with interface consistency. :smile: